testing/integration/beat_receivers_test.go

```diff
@@ -7,180 +7,462 @@
+			{dsType: "metrics", dsDataset: "elastic_agent.filebeat_input", dsNamespace: info.Namespace},
+			{dsType: "metrics", dsDataset: "elastic_agent.metricbeat", dsNamespace: info.Namespace},
+		}
+
+		for _, tc := range tests {
+			require.Eventuallyf(t,
+				func() bool {
+					findCtx, findCancel := context.WithTimeout(context.Background(), 10*time.Second)
+					defer findCancel()
+					rawQuery := map[string]any{
+						"query": map[string]any{
+							"bool": map[string]any{
+								"must": []map[string]any{
+									{
+										"match": map[string]any{"data_stream.type": tc.dsType},
+									},
+									{
+										"match": map[string]any{"data_stream.dataset": tc.dsDataset},
+									},
+									{
+										"match": map[string]any{"data_stream.namespace": tc.dsNamespace},
+									},
 								},
 							},
 						},
-					},
+					}
+					docs, err := estools.PerformQueryForRawQuery(findCtx, rawQuery, tc.dsType+"-*", info.ESClient)
+					require.NoError(t, err)
+					return docs.Hits.Total.Value > 0
+				},
+				2*time.Minute, 5*time.Second,
+				"No documents found for type: %s, dataset: %s, namespace: %s", tc.dsType, tc.dsDataset, tc.dsNamespace)
+		}
+
+		// get self monitoring logs
+		var agentDocs estools.Documents
+		fbMonitoringIndex := "logs-elastic_agent-" + info.Namespace
+		require.Eventually(t,
+			func() bool {
+				findCtx, findCancel := context.WithTimeout(ctx, 10*time.Second)
+				defer findCancel()
+
+				agentDocs, err = estools.GetLogsForIndexWithContext(findCtx, info.ESClient, ".ds-"+fbMonitoringIndex+"*", map[string]interface{}{
+					"message": commonLogMessage,
+				})
+				require.NoError(t, err)
+
+				if agentDocs.Hits.Total.Value != 0 {
+					logsEADocs = agentDocs
+					return true
 				}
-				docs, err := estools.PerformQueryForRawQuery(findCtx, rawQuery, tc.dsType+"-*", info.ESClient)
+				return false
+			},
+			2*time.Minute, 1*time.Second, "could not find monitoring log")
+	})
+
+	t.Run("compare logs ingested by agent monitoring vs otel monitoring", func(t *testing.T) {
+		if runtime.GOOS == "windows" {
+			t.Skip("skipping this test on windows for now")
+		}
+
+		// Not proceed with this test if monitoring logs from elastic-agent does not exist
+		require.NotPanics(
+			t, func() {
+				_ = logsEADocs.Hits.Hits[0].Source
+			}, "monitoring logs from elastic-agent should exist before proceeding",
+		)
+
+		fbReceiverMonitoringIndex := "logs-elastic_agent-monitoringotel"
+
+		type configOptions struct {
+			InputPath      string
+			HomeDir        string
+			ESEndpoint     string
+			ESApiKey       string
+			BeatsESApiKey  string
+			SocketEndpoint string
+		}
+		esEndpoint, err := getESHost()
+		require.NoError(t, err, "error getting elasticsearch endpoint")
+		esApiKey, err := createESApiKey(info.ESClient)
+		require.NoError(t, err, "error creating API key")
+		require.NotEmptyf(t, esApiKey.Encoded, "api key is invalid %q", esApiKey)
+
+		var inputPath string
```

Comment:
I really don't like the inputPath based on GOOS.  Once it is installed `fixture.WorkDir()` should point to the correct top directory.  I wonder if we should first install with a blank `elastic-agent.yml`, then write the correct `elastic-agent.yml` and then do a restart.  That way we would know the correct paths.  We can use a filter on our searches based on time to exclude any logs that might be generated with the blank `elastic-agent.yml`